Resources/doc/pipeline.rst

```diff
@@ -0,0 +1,40 @@
+Pipeline
+========
+
+Pipeline is used to process data with 5 events:
+
+ - source
+ - start
+ - modify
+ - consume
+ - finish
+
+Pipeline starts with source event which provides all data which should be processed.
+Then start event is fired to indicate that items are about to come
+after that pipeline iterates through all items from all sources by calling modify and consume events
+with each item. After the iteration finish event is fired to notify that no more items will follow.
```

Comment:
After the iterations are finished, finish event is fired to notify that no more items will follow.